sdk/trace/src/main/scala/org/typelevel/otel4s/SpanContextImpl.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s // ?
+
+import org.typelevel.otel4s.trace.SpanContext
+
+final class SpanContextImpl extends SpanContext { // does a impl belong in the sdk?
```

Comment:
I _think_ we can make this a `case class` with all the fields below as its members. And actually we can probably keep the class `private`, since users will always work with the `SpanContext` interface.